docs/DeveloperGuide.md

```diff
@@ -1,33 +1,373 @@
+bookmark, timetable, academic and flashcard. Each of these component
+provide different functionalities, and they combine to form our software -- Study It.
+
+Component | What it does
+----------|----------------
+Bookmark  | Stores internet links under different categories for easier access
+Timetable | Tracks and display your weekly schedule
+Academic  | Stores important contacts and keep track of your grades
+Flashcard | Stores study questions and allow you to quiz yourself for easier memorization
+
+The **main component** of Study It helps user access each of these components.
+It processes the user input, provides general functionalities to traverse the software and other
+helpful functionalities. This will be further explained under Study It's design later.
+
+### **Architecture**
+This section describes the overall architecture of Study It.
+
+![Architecture Diagram](Images/ArchitectureDiagrams/architecture.png)
+
+The *Architecture Diagram* given above explains the high-level design of the software.
+Each component in the diagram will be briefly explained below.
+
+The **`main`** in **`Study It`** is responsible for initializing all the other components
+in the software.
+
+The following is a brief overview of the components in Study It:
+* **UI**: Handles the communication between the software and the user
+* **Parser**: Processes the input from the user
+* **Execution**: Perform the actions determined by the parser
+
+The following components processes their respective actions (parsing and execution) and handles their own
+storage component: 
+* **Bookmark**
+* **Timetable**
+* **Academic**
+* **Flashcard**
+
+**General commands** are commands that takes priority in execution no matter which mode the software is currently
+in.
+
+**Modes of the program**
+
+Study It has 5 **modes** that the user can navigate between:
+* **Main menu**: Default mode when booting the software, there are no functionalities tied to this mode
+* **Bookmark**
+* **Timetable**
+* **Academic**
+* **Flashcard**
+
+Each of these modes has their own unique commands. The user must switch to that mode to perform the
+mode's functionalities.
+
+User can switch between these modes at any point when using the software.
+
+**How the architecture components interact with each other**
+
+![Sequence Overview](Images/ArchitectureDiagrams/overviewsequence.png)
+
+As shown in the sequence diagram above, the user will interact with Study It via the User Interface (UI).
+The commands received will then be parsed to identify the type of command. If the command is one of the 
+general commands, it'll be processed and provide output. Otherwise, the command will be relayed
+to different components to be processed based on the software's current mode. 
+
+### **Main Component**
+This section will give an overview of the main component of Study It
+
+![Main Component Diagram](Images/MainComponentDiagrams/maincomponent.png)
```

Comment:
Diagram is quite difficult to read. Maybe a font change or size increase could increase the readability?